### PR TITLE
Add Support for OuDiaSecond (`oud2`) File Extension

### PR DESCRIPTION
**Pull request by yocjyet**
## TL;DR

This PR adds OuDiaSecond (`.oud2`) file format support.

## Background

[OuDiaSecond](http://oudiasecond.seesaa.net/) is a famous OuDia successor that is now widely used for the same purpose. There is a lot of changes especially it adds a lot keys and configurations. However, the file format itself (`.oud2`) does not differ a lot from `.oud` file. The structures are the same, so the same VSCode language extension `oud-intellisense` can be used directly to add support for OuDiaSecond file. However, since this format is not directly supported by this plugin, user must add the following configuration to their settings.

```json
"files.associations": {
  "*.oud2": "oudia"
}
```

## Motivation
Supporting this format ensures that the VSCode extension remains up-to-date with current trends in diagramming software.

## Changes
- Added support for `.oud2` by add the file extension to `extensions` in [`package.json`](https://github.com/up-tri/oud-intellisense/compare/master...yocjyet:oud-intellisense:master#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519)
- Updated the [`README.md`](https://github.com/up-tri/oud-intellisense/compare/master...yocjyet:oud-intellisense:master#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5) to reflect the addition of `.oud2` support. 

## Backward Compatibility
These changes are backward compatible with the existing `.oud` format and should not affect any current users.

Please let me know if any adjustments are needed. I am happy to iterate on this PR. Thank you for your time and consideration!

### CODE REVIEW

**Review by up-tri — APPROVED** [review]

**Comment by up-tri** [comment]
@yocjyet Thank you so much for your contribution!
If there are any further improvements or updates in the future, feel free to contribute again.
